pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/lineage/DefaultLineageManager.java

```diff
@@ -105,6 +107,11 @@ public void updateLineageForRetention(TableConfig tableConfig, SegmentLineage li
     }
   }
 
+  @Override
+  public List<List<SegmentZKMetadata>> getSegmentGroupsForMergeRollupTask(TableConfig tableConfig,
```

Comment:
I feel that this function doesn't belong to `LineageManager`? LineageManager should not know about `MergeRollupTask`. MergeRollupTask should be the user of the segment replace protocol.

Can you add a bit more information on what it means by `SegmentGroup` here?

If your intention is to inject the custom code to the MergeRollupTask, please create a dedicated interface for that.

Reply:
+1
This logic belongs to MergeRollup, not LineageManager. Let's create an interface like MergeRollupSegmentGroupManager, MergeRollupSegmentCategorizer, or something like these, and inject the interface into MergeRollupTaskGenerator.

Reply:
@snleee @sajjad-moradi  Thanks for the suggestion, removed from `LineageManager` and added `PinotTaskSegmentGroupManager` interface. Custom class can be registered via `PinotTaskManager.registerTaskSegmentGroupManager()`. Please take another look.